modules/platform/nuxeo-platform-default-config/src/main/resources/OSGI-INF/document-pageprovider-contrib.xml

```diff
@@ -136,6 +136,25 @@
       <pageSize>20</pageSize>
     </coreQueryPageProvider>
 
+    <genericPageProvider class="org.nuxeo.elasticsearch.provider.ElasticSearchNxqlPageProvider" name="picker">
```

Comment:
Tests are failing because `nuxeo-platform-default-config` is deployed on some tests but not `nuxeo-elasticsearch-*` modules.

Moreover, you don't need a new `Picker` document type as it contains only a `title` field, there is already a title field in the `AdvancedSearch` doc type.

The page provider is linked to the `Picture` facet, so I would move it to `nuxeo-platform-imaging-core` module in a new contribution file, and not `nuxeo-platform-default-config`.

I would have declared it like the other page providers (not strictly linked to ES as it's not needed):
```
  <coreQueryPageProvider name="picker">
      <trackUsage>true</trackUsage>
      <whereClause docType="AdvancedSearch">
        <predicate parameter="dc:title" operator="FULLTEXT">
          <field schema="advanced_search" name="title" />
        </predicate>
        <fixedPart>
          ecm:mixinType = 'Picture' AND
          file:content IS NOT NULL AND
          ecm:mixinType != 'HiddenInNavigation' AND
          ecm:isVersion = 0 AND
          ecm:isTrashed = 0
        </fixedPart>
      </whereClause>
      <pageSize>20</pageSize>
  </coreQueryPageProvider>
```

An then, add it to the list here https://github.com/nuxeo/nuxeo/blob/e2150ea02a11905fe564315d7e91036153cb6476/server/nuxeo-nxr-server/src/main/resources/templates/common-base/nuxeo.defaults#L105 so that it can be used with ES.

@bdelbosc do you confirm that's the good approach?

Reply:
It cannot be generic as `file:content IS NOT NULL` only works on ES.
Adding things to `elasticsearch.override.pageproviders` when it can be declared in the XML instead is weird. The property is for overrides, we shouldn't use it as part of the general config.

Reply:
Note for the future: if we had introduced a new document type / schema I would have insisted that it be called `NuxeoFilePicker` and same for the schema, as there's a risk of collision with customer types otherwise.